app/controllers/topics_controller.rb

```diff
@@ -543,6 +543,7 @@ def merge_topic
     guardian.ensure_can_move_posts!(topic)
 
     dest_topic = topic.move_posts(current_user, topic.posts.pluck(:id), destination_topic_id: params[:destination_topic_id].to_i)
+    DiscourseEvent.trigger(:topic_merged, topic, dest_topic)
```

Comment:
I support this new event but it should not be triggered in the controller, it should be triggered in topic.move_posts

Also no need to sign in and test the controller then cause this will always happen. 

Closing this for now, feel free to re-submit